pytorch_lightning/callbacks/progress.py

```diff
@@ -43,6 +42,35 @@ class tqdm(_tqdm):
     Custom tqdm progressbar where we append 0 to floating points/strings to prevent the progress bar from flickering
     """
 
+    def __init__(self, *args, **kwargs):
+        self.using_text_io_wrapper = False
+        super().__init__(*args, **kwargs)
+
+        self.using_text_io_wrapper = hasattr(self, "fp") and isinstance(self.fp, io.TextIOWrapper)
+
+    def display(self, msg=None, pos=None):
```

Comment:
Can you explain a bit what this does and how it does it?

Why do we need all the if/else instead of:

```python
super().display(msg=msg, pos=pos)
if self.using_textio_wrapper and self.n == self.total:
    fix_file_lines()
```